### ISSUE TITLE
Reproducing results with 1024 points on ModelNet40

### ISSUE THREAD

**Issue by haotian-liu** [lulutang0608/Point-BERT#5]
Hi, thank you first for your great work Point-BERT!  I am trying to running your code to reproduce the results with 1024 points on ModelNet40 (93.2% acc w/ voting).

I fine tune Point-BERT on ModelNet40 (initialized the [provided weights](https://cloud.tsinghua.edu.cn/f/202b29805eea45d7be92/?dl=1) in your repo pretrained on ShapeNetPart).  I am getting 92.26% accuracy with voting.  I also evaluate with your [model weights](https://cloud.tsinghua.edu.cn/f/9be5d9dcbaeb48adb360/?dl=1) on ModelNet40, and can get 93.2% accuracy.  Could you please advise if there is anything I did wrong in **training (fine-tuning)** on ModelNet40?  Thank you!

**Comment by yuxumin** [comment]
Thanks for your interest in our work.

Could you please provide more details in your reproducing experiments? The pretrained model is trained with one V100 with torch 1.8.0;
This is the training log for provided model: [log](https://github.com/lulutang0608/Point-BERT/files/7698042/PointTransformer_1024points.log). You can check the config and some other details in this log file.

Best!

**Comment by haotian-liu** [comment]
Thank you for your quick reply and for sharing these details!  I will try to compare and see if there are any differences.

As a separate note, I noticed that `moco_loss` is set to `False` [here](https://github.com/lulutang0608/Point-BERT/blob/master/cfgs/Mixup_models/Point-BERT.yaml#L36), I wonder (1) whether the pretrained weights are trained with moco_loss or not; (2) whether we need to turn on MoCo loss when we pretrain the Point-BERT for the best performance (is it the same as Moco column in Table 5a)?

Thank you so much!

**Comment by yuxumin** [comment]
(1) `moco_loss`  is set to `False` during the pre-training
(2) Sorry for confusion. `moco_loss` is not the same as MoCo column in Table 5a. As you can see from [line](https://github.com/lulutang0608/Point-BERT/blob/master/models/Point_BERT.py#L542), `moco_loss` is the constrastive loss between origin samples and the memory bank. In Table 5a, MoCo column means we optimize the contrastive loss between mixed samples and the memory bank, which is `cutmix_loss` in config.

We will correct those names to avoid more confusion.

**Comment by haotian-liu** [comment]
Thank you for the clarification.  I found one difference was that I was using distributed training instead of single-gpu training.  I tried two runs with a single RTX 3090, the accruacy was 92.26%/92.67%, and the voting accruacy was 92.67%/92.46%.  I am using PyTorch 1.10, and the command I am using was `CUDA_VISIBLE_DEVICES=0 python main_BERT.py --config cfgs/ModelNet_models/PointTransformer.yaml --exp_name=pointBERT_finetune_ModelNet_pretrainBERT_singlegpu_run1 --finetune_model --ckpts experiments/pretrained/Point-BERT.pth --val_freq 10`, where `Point-BERT.pth` is downloaded from [here](https://cloud.tsinghua.edu.cn/f/202b29805eea45d7be92/?dl=1).  I might also try downgrading my PyTorch version to 1.8 to see if it is the cause, but meanwhile, would you kindly advise anything you think that could be wrong?  Thank you!

**Comment by yuxumin** [comment]
The environment of pytorch does not seem to be a problem.  
I think the `val_freq` should not be set to 10 in ModelNet training, using default setting of `val_freq = 1` is better.
As for other things that may cause a drop of performance, I get no ideas now. I check my training log on ModelNet, the worse running of the model get an accuracy of 92.91 (voting). Haven't you changed the config file, right?

**Comment by haotian-liu** [comment]
@yuxumin Thank you for your quick reply.
Regarding the config, I did not make any change to the config file, and the updated results using PyTorch 1.8.0 is trained with a completely new repo clone so I can make sure that it has never been modified.
And thank you for pointing out the `val_freq`, could you please elaborate a bit more on why `val_freq` matters? To my understanding, that is just for controlling how often we evaluate when we train the model, and it should not affect the final trained model.  Please correct me if I am wrong or there is anything that I am missing.
Thank you so much!

**Comment by yuxumin** [comment]
We follow the previous work ([PointNet](https://github.com/yanx27/Pointnet_Pointnet2_pytorch/blob/master/train_classification.py), [PointNet++](https://github.com/yanx27/Pointnet_Pointnet2_pytorch/blob/master/train_classification.py), [DGCNN](https://github.com/WangYueFt/dgcnn/blob/master/pytorch/main.py)) to evaluate the model for every epoch and some previous works, such as [RSCNN](https://github.com/Yochengliu/Relation-Shape-CNN/blob/master/train_cls.py), even evaluate the model  more frequently to choose the best model. So, the `val_freq` may influence the final performance. We just follow the previous works, and haven't tried other evaluation scheme.

**Comment by haotian-liu** [comment]
Thank you so much for the clarification!
